### PR TITLE
add cncf-landscape shortcode for training page

### PR DESCRIPTION
**Pull request by Arhell**
issue https://github.com/kubernetes/website/issues/23911
/area web-development
/language en

### CODE REVIEW

File: layouts/shortcodes/cncf-landscape.html [hunk 1]
```diff
@@ -56,8 +56,15 @@
 });
 </script>
 {{- end -}}
+{{ if in (.Site.BaseURL | string) "https://kubernetes.io/partners" }}
```
Comment (sftim) [inline layouts/shortcodes/cncf-landscape.html:59]: You shouldn't need to match on the URL where the shortcode is used. Instead, add `data-*` elements to the page where you're using it.

Comment (sftim) [reply layouts/shortcodes/cncf-landscape.html:59]: Maybe the shortcode could take a new parameter, though, `category` (eg `kubernetes-training-partner`).

Comment (kbhawkey) [reply layouts/shortcodes/cncf-landscape.html]: What are the possible categories?

**Comment by kbhawkey** [comment]
Page preview?
https://deploy-preview-24014--kubernetes-io-master-staging.netlify.app/training/

**Comment by Arhell** [comment]
> Page preview?
> https://deploy-preview-24014--kubernetes-io-master-staging.netlify.app/training/

yes

File: layouts/shortcodes/cncf-landscape.html [hunk 2]
```diff
@@ -54,6 +54,25 @@
...
 });
+
+var documentReady = function (callback) {
+  if (document.readyState !== "loading") callback();
+  else document.addEventListener("DOMContentLoaded", callback);
+};
+
+documentReady(function () {
+  function checkPage() {
+    var trainingPage = "https://landscape.cncf.io/category=kubernetes-training-partner&format=logo-mode&grouping=category&embed=yes";
```
Comment (sftim) [inline layouts/shortcodes/cncf-landscape.html:65]: This hard-codes a specific category. I don't think any part of this shortcode should hard code category names.
Instead, take those as parameters.

File: layouts/shortcodes/cncf-landscape.html [hunk 3]
```diff
@@ -54,6 +54,25 @@
...
+    var iframeId = document.getElementById("landscape");
+    var checkClass = document.getElementsByClassName("landscape-section");
+    if (checkClass.length) {
+      iframeId.src = trainingPage
```
Comment (sftim) [inline layouts/shortcodes/cncf-landscape.html:69]: This is hard coding a detail that should be available from context instead.

**Comment by sftim** [comment]
Just a note, I've got less time than I would typically have to help with early and indeed formal PR review at the moment - so, whilst I'm happy to chip in, that feedback might have to be more terse than I'd like it to be.

**Comment by Arhell** [comment]
I updated a little, but here the iframe is displayed without any buttons (as in the example issue), alas, there is a hard code

File: layouts/shortcodes/cncf-landscape.html [hunk 4]
```diff
@@ -53,7 +53,19 @@
   window.addEventListener("hashchange", hashChangeHandler, false);
   /* handle initially-set fragment */
   hashChangeHandler();
+
+  function checkUrl() {
+    let windowLocation = window.location.pathname;
+    var urlName = 'kubernetes-training-partner';
```
Comment (habibrosyad) [inline layouts/shortcodes/cncf-landscape.html:59]: Is it possible for this to be in the parameter?

File: layouts/shortcodes/cncf-landscape.html [hunk 5]
```diff
@@ -53,7 +53,19 @@
...
+    if (windowLocation == '/training/') {
```
Comment (habibrosyad) [inline layouts/shortcodes/cncf-landscape.html:60]: also this...
maybe at the end the shortcode can be called like this:
```
{{< cncf-landscape helpers=true category="kubernetes-training-partner" check-path="/training/">}}
```
and if `checkUrl` is actually only needs to be done in certain path, you could just avoid making this function and instead do something like this:

```js
{{ $checkPath := .Get "check-path" }}
{{ if $checkPath }}
if (window.location.pathname === "{{ $checkPath }}") {
  var landscapeElement = document.getElementById("landscape");
  landscapeElement.src = "https://landscape.cncf.io/category="+"{{ .Get "category" | default "some-default-cat" }}"+"&format=card-mode&grouping=category&embed=yes";
}
{{ end }}
```

**Review by habibrosyad — COMMENTED** [review]
Just a few suggestions...

File: layouts/shortcodes/cncf-landscape.html (hunk 5 above)
Comment (habibrosyad) [reply layouts/shortcodes/cncf-landscape.html:60]: hmm... my confusion is do we really need to check whether we are in `/training/` to run this?

Comment (sftim) [reply layouts/shortcodes/cncf-landscape.html:60]: > ```
> {{< cncf-landscape helpers=true category="kubernetes-training-partner" check-path="/training/">}}
> ```

Along those lines, yes. I'd had in mind something more like:
```
{{< cncf-landscape helpers=false category="kubernetes-training-partner" >}}
```

Comment (habibrosyad) [reply layouts/shortcodes/cncf-landscape.html:60]: It's even simpler then, no need to check for the path (which I have doubt to be required in this case) 😄

File: content/en/training/_index.html [hunk 6]
```diff
@@ -112,7 +112,6 @@ <h2>Kubernetes Training Partners</h2>
     </center>
   </div>
   <div class="main-section landscape-section">
-    <iframe src="https://landscape.cncf.io/category=kubernetes-training-partner&format=logo-mode&grouping=category&embed=yes" frameborder="0" id="landscape" scrolling="no"></iframe>
-    <script src="https://landscape.cncf.io/iframeResizer.js"></script>
+    {{< cncf-landscape helpers=false $training=true >}}
```
Comment (sftim) [inline content/en/training/_index.html:115]: @Arhell this looks like it is hard-coding a detail of the training page

How about making the call be:
```markdown
     {{< cncf-landscape helpers=false category="kubernetes-training-partner" >}}
```
and then adjusting the shortcode to use that data?

Comment (Arhell) [reply content/en/training/_index.html]: done, check please 
review page
https://deploy-preview-24014--kubernetes-io-master-staging.netlify.app/training/

File: layouts/shortcodes/cncf-landscape.html [hunk 7]
```diff
@@ -54,10 +54,18 @@
...
   hashChangeHandler();
 });
+
 </script>
-{{- end -}}
+{{ end }}
+{{ $category := .Get "category" }}
+{{ if eq $category "kubernetes-training-partner" }}
+<iframe src="https://landscape.cncf.io/category=kubernetes-training-partner&format=logo-mode&grouping=category&embed=yes" frameborder="0" id="landscape" scrolling="no"></iframe>
+<script src="https://landscape.cncf.io/iframeResizer.js"></script>
```
Comment (sftim) [inline layouts/shortcodes/cncf-landscape.html:63]: I plan to suggest a tweaked version of this. I haven't time right now - sorry.

Comment (sftim) [reply layouts/shortcodes/cncf-landscape.html:63]: (what I want to do is copy the category from the parameter into the HTML)

Comment (kbhawkey) [reply layouts/shortcodes/cncf-landscape.html:63]: OK. I have not been following these changes, but you want to populate the `src` attribute with
the `category` variable? Something like:
```
<iframe src="https://landscape.cncf.io/category={{$category}}&format=logo-mode&grouping=category&embed=yes" frameborder="0" id="landscape" scrolling="no"></iframe>
```

File: layouts/shortcodes/cncf-landscape.html [hunk 8]
```diff
@@ -54,10 +54,18 @@
...
+{{ else }}
+
 <div id="frameHolder">
   <iframe frameborder="0" id="landscape" scrolling="no" src="" style="width: 1px; min-width: 100%" title="CNCF Landscape"></iframe>
   <script src="https://landscape.cncf.io/iframeResizer.js"></script>
 </div>
-{{- end -}}
+{{ end }}
+{{ end }}
```
[the remark this answers is not in the document]
Comment (sftim) [reply layouts/shortcodes/cncf-landscape.html:71]: (guess who forgot to press Send) :wink:

[the remark this answers is not in the document]
Comment (Arhell) [reply layouts/shortcodes/cncf-landscape.html:71]: but on the page https://kubernetes.io/training/
without <div id = "frameHolder">
or let it be?

[the remark this answers is not in the document]
Comment (sftim) [reply layouts/shortcodes/cncf-landscape.html:71]: Oh, well spotted I didn't notice that difference. I think the `frameHolder` id is OK everywhere.

File: content/en/training/_index.html [hunk 9]
```diff
@@ -112,7 +112,6 @@ <h2>Kubernetes Training Partners</h2>
     </center>
   </div>
   <div class="main-section landscape-section">
-    <iframe src="https://landscape.cncf.io/category=kubernetes-training-partner&format=logo-mode&grouping=category&embed=yes" frameborder="0" id="landscape" scrolling="no"></iframe>
-    <script src="https://landscape.cncf.io/iframeResizer.js"></script>
+    {{< cncf-landscape helpers=false category >}}
```
Comment (sftim) [inline content/en/training/_index.html:115]:
```suggestion
    {{< cncf-landscape helpers="false" category="kubernetes-training-partner" >}}
```

ought to do it

Comment (Arhell) [reply content/en/training/_index.html]: Thanks!
@sftim @kbhawkey 
revirew page 
https://deploy-preview-24014--kubernetes-io-master-staging.netlify.app/training/

File: layouts/shortcodes/cncf-landscape.html [hunk 10]
```diff
@@ -57,7 +57,11 @@
 </script>
 {{- end -}}
 <div id="frameHolder">
-  <iframe frameborder="0" id="landscape" scrolling="no" src="" style="width: 1px; min-width: 100%" title="CNCF Landscape"></iframe>
+  {{ if ( .Get "category" ) }}
+  <iframe frameborder="0" id="landscape" scrolling="no" src="https://landscape.cncf.io/category={{ .Get "category" }}&format=logo-mode&grouping=category&embed=yes"></iframe>
+  {{ else }}
+  <iframe frameborder="0" id="landscape" scrolling="no" src="https://landscape.cncf.io/format=logo-mode;embed=yes" style="width: 1px; min-width: 100%" title="CNCF Landscape"></iframe>
```
Comment (kbhawkey) [inline layouts/shortcodes/cncf-landscape.html:63]: The rendered page looks nice.
I am not familiar with the various values of `format`. What other options are there besides `logo-mode`?
Is there an example in the docs of a page that shows the entire landscape (else clause)?
What is the point of the `helpers` attribute?

Comment (kbhawkey) [reply layouts/shortcodes/cncf-landscape.html:63]: I added this updated shortcode to a content page. The landscape logos do not use up the page as expected (width).
I'd like the shortcode to be used without adding extra HTML to the Markdown page.

**Pull request merged**